src/System.Xml.XmlSerializer/tests/XmlSerializerTests.cs

```diff
@@ -1111,6 +1111,17 @@ public static void Xml_TypeWithMismatchBetweenAttributeAndPropertyType()
         Assert.StrictEqual(value.IntValue, actual.IntValue);
     }
 
+    [Fact]
+    public static void Xml_TypeWithNonPublicDefaultConstructor()
```

Comment:
I haven't looked at all of the tests not covered by this PR, so these may already exist, but do you have tests that cover:
- when the find/use ConstructorInfo code is used
- when Activator.CreateInstance is used
- when neither works?